src/main/java/com/example/backend/RestController/RoomController.java

```diff
@@ -85,4 +87,18 @@ public Message updateReservation(@PathVariable long id) {
          repo.deleteById(id);
          return new Message("room has been deleted");
      }
-}
+     
+     @GetMapping("/{location}/{numGuest}/{booked}")
+     public List<Room> fetchRoom(@PathVariable String location, @PathVariable int numGuest, @PathVariable boolean booked) {
+    	 Iterable<Room> data = repo.findAll();
+    	 List<Room> rooms = new ArrayList<>();
+    	 Room room = new Room(location.replace('-', ' '), numGuest, booked); //replace hypens if there are any
+    	 
+    	 for (Room r : data) {
+    		 if (r.sameRoom(room)) {
+    			 rooms.add(r);
+    		 }
+    	 }
+    	 return rooms;
```

Comment:
You can look into RequestParameter, as used in ReservationController. The space can be replaced by %20. For example, http://localhost:8080/reservation/test?whatever=hello%20fds

Reply:
Yeah, I added that to the fetch method now. I tried spaces and apostrophes in the URL and it seems like Spring takes care of the spaces and other characters automatically.